aws-logs-logstream/src/main/java/software/amazon/logs/logstream/BaseHandlerStd.java

```diff
@@ -0,0 +1,136 @@
+private final static String RESOURCE_NOT_FOUND_EXCEPTION = "ResourceNotFoundException";
+
+
+  protected BaseHandlerStd() {
+    this(ClientBuilder.getClient());
+  }
+
+  protected BaseHandlerStd(CloudWatchLogsClient cloudWatchLogsClient) {
+    this.cloudWatchLogsClient = requireNonNull(cloudWatchLogsClient);
+  }
+
+  private CloudWatchLogsClient getCloudWatchLogsClient() {
+    return cloudWatchLogsClient;
+  }
+
+  @Override
+  public final ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final Logger logger) {
+    return handleRequest(
+      proxy,
+      request,
+      callbackContext != null ? callbackContext : new CallbackContext(),
+      proxy.newProxy(ClientBuilder::getClient),
+      logger
+    );
+  }
+
+  protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+    final AmazonWebServicesClientProxy proxy,
+    final ResourceHandlerRequest<ResourceModel> request,
+    final CallbackContext callbackContext,
+    final ProxyClient<CloudWatchLogsClient> proxyClient,
+    final Logger logger);
+
+  protected ProgressEvent<ResourceModel, CallbackContext> doesResourceExistwithName(final ProxyClient<CloudWatchLogsClient> proxyClient, final ResourceModel model, final CallbackContext callbackContext)
+          throws AwsServiceException {
+    final DescribeLogStreamsRequest translateToReadRequest = Translator.translateToReadRequest(model);
+    final DescribeLogStreamsResponse response;
+    try {
+      response = proxyClient.injectCredentialsAndInvokeV2(translateToReadRequest, proxyClient.client()::describeLogStreams);
+      if(CollectionUtils.isEmpty(response.logStreams())){
+        // Log Stream does not exist so return InProgress Event
+        return ProgressEvent.progress(model, callbackContext);
+      }
+      final LogStream logStream = response.logStreams().get(0);
+      if(logStream.logStreamName().equals(model.getLogStreamName())){
+        // Log Stream does exist so return Failed
+        return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.AlreadyExists, String.format("Log Stream Already Exists"));
+      }
+      return ProgressEvent.progress(model, callbackContext);
+    } catch (final AwsServiceException e) {
+      BaseHandlerException newException = Translator.translateException(e);
+      if (newException instanceof CfnNotFoundException) {
+        return ProgressEvent.progress(model, callbackContext);
+      }
+      throw e;
+    }
+  }
+
+  protected void logExceptionDetails(Exception e, Logger logger, final String stackId) {
+    logger.log(String.format("Stack with ID: %s got exception: %s Message: %s Cause: %s", stackId,
+            e.toString(), e.getMessage(), e.getCause()));
+  }
+
+  protected void handleException(Exception e, Logger logger, final String stackId) {
```

Comment:
Why have two different methods doing the same thing? handleException and handleError.

Reply:
removed handleException